tests/checkers/unittest_type_annotations.py

```diff
@@ -0,0 +1,271 @@
+            """
+        from abc import abstractmethod
+
+        class MyClass:
+            @abstractmethod
+            def foo(self, x):  #@
+                pass
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_property_skipped(self) -> None:
+        """Test that property methods are skipped."""
+        node = astroid.extract_node(
+            """
+        class MyClass:
+            @property
+            def foo(self):  #@
+                return 42
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_vararg_missing_annotation(self) -> None:
+        """Test detection of missing *args annotation."""
+        node = astroid.extract_node(
+            """
+        def foo(*args) -> None:  #@
+            pass
+        """
+        )
+        with self.assertAddsMessages(
+            MessageTest(
+                msg_id="missing-param-type-annotation",
+                args=("args", "foo"),
+                node=node,
+                line=2,
+                col_offset=0,
+                end_line=2,
+                end_col_offset=7,
+            )
+        ):
+            self.checker.visit_functiondef(node)
+
+    def test_kwarg_missing_annotation(self) -> None:
+        """Test detection of missing **kwargs annotation."""
+        node = astroid.extract_node(
+            """
+        def foo(**kwargs) -> None:  #@
+            pass
+        """
+        )
+        with self.assertAddsMessages(
+            MessageTest(
+                msg_id="missing-param-type-annotation",
+                args=("kwargs", "foo"),
+                node=node,
+                line=2,
+                col_offset=0,
+                end_line=2,
+                end_col_offset=7,
+            )
+        ):
+            self.checker.visit_functiondef(node)
+
+    def test_fully_annotated_with_varargs(self) -> None:
+        """Test that fully annotated functions with *args and **kwargs work."""
+        node = astroid.extract_node(
+            """
+        def foo(x: int, *args: str, **kwargs: bool) -> None:  #@
+            pass
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_keyword_only_args_missing_annotation(self) -> None:
+        """Test detection of missing keyword-only argument annotations."""
+        node = astroid.extract_node(
+            """
+        def foo(x: int, *, y) -> None:  #@
+            pass
+        """
+        )
+        with self.assertAddsMessages(
+            MessageTest(
+                msg_id="missing-param-type-annotation",
+                args=("y", "foo"),
+                node=node,
+                line=2,
+                col_offset=0,
+                end_line=2,
+                end_col_offset=7,
+            )
+        ):
+            self.checker.visit_functiondef(node)
```

Comment:
Great tests (probably). Functional tests are a lot easier to review and write (checking the line number in particular which I didn't do here). Those are faster to run though so let's not rewrite everything unless another maintainer thinks we should.

Reply:
We either need to add typing in the code example or disable in the command(s?) used. If there's a generic command it might be less work to add a disable